Binary files com.discord-1078-base/build/apk/classes.dex and com.discord-1078/build/apk/classes.dex differ
Binary files com.discord-1078-base/dist/com.discord-1078.apk and com.discord-1078/dist/com.discord-1078.apk differ
diff -crB com.discord-1078-base/smali/com/discord/stores/StoreExperiments$getExperimentalAlpha$1.smali com.discord-1078/smali/com/discord/stores/StoreExperiments$getExperimentalAlpha$1.smali
*** com.discord-1078-base/smali/com/discord/stores/StoreExperiments$getExperimentalAlpha$1.smali	2020-03-26 10:35:54.852235186 +0300
--- com.discord-1078/smali/com/discord/stores/StoreExperiments$getExperimentalAlpha$1.smali	2020-03-27 12:53:46.243461585 +0300
***************
*** 80,110 ****
  
  .method public final call(Lcom/discord/models/domain/ModelUser;Lcom/discord/models/domain/ModelGuild;)Z
      .locals 1
  
!     const-string v0, "meUser"
! 
!     .line 2
!     invoke-static {p1, v0}, Ly/u/b/j;->checkExpressionValueIsNotNull(Ljava/lang/Object;Ljava/lang/String;)V
! 
!     invoke-virtual {p1}, Lcom/discord/models/domain/ModelUser;->isStaff()Z
! 
!     move-result p1
! 
!     if-nez p1, :cond_1
! 
!     if-eqz p2, :cond_0
! 
!     goto :goto_0
! 
!     :cond_0
!     const/4 p1, 0x0
! 
!     goto :goto_1
! 
!     :cond_1
!     :goto_0
!     const/4 p1, 0x1
! 
!     :goto_1
!     return p1
  .end method
--- 80,86 ----
  
  .method public final call(Lcom/discord/models/domain/ModelUser;Lcom/discord/models/domain/ModelGuild;)Z
      .locals 1
+     const/4 v0, 0x1
  
!     return v0
  .end method
diff -crB com.discord-1078-base/smali/com/discord/stores/StoreExperiments.smali com.discord-1078/smali/com/discord/stores/StoreExperiments.smali
*** com.discord-1078-base/smali/com/discord/stores/StoreExperiments.smali	2020-03-26 10:35:54.862235237 +0300
--- com.discord-1078/smali/com/discord/stores/StoreExperiments.smali	2020-03-27 12:53:46.243461585 +0300
***************
*** 1318,1375 ****
  .end method
  
  .method public final isExperimentalAlpha$app_productionDiscordExternalRelease()Z
!     .locals 4
  
!     .line 1
!     iget-boolean v0, p0, Lcom/discord/stores/StoreExperiments;->initialized:Z
  
!     const/4 v1, 0x1
! 
!     if-eqz v0, :cond_0
! 
!     .line 2
!     iget-object v0, p0, Lcom/discord/stores/StoreExperiments;->storeUser:Lcom/discord/stores/StoreUser;
! 
!     invoke-virtual {v0}, Lcom/discord/stores/StoreUser;->getMeInternal$app_productionDiscordExternalRelease()Lcom/discord/models/domain/ModelUser$Me;
! 
!     move-result-object v0
! 
!     if-eqz v0, :cond_0
! 
!     invoke-virtual {v0}, Lcom/discord/models/domain/ModelUser;->isStaff()Z
! 
!     move-result v0
! 
!     if-eq v0, v1, :cond_2
! 
!     .line 3
!     :cond_0
!     iget-object v0, p0, Lcom/discord/stores/StoreExperiments;->storeGuilds:Lcom/discord/stores/StoreGuilds;
! 
!     invoke-virtual {v0}, Lcom/discord/stores/StoreGuilds;->getGuildsInternal$app_productionDiscordExternalRelease()Ljava/util/Map;
! 
!     move-result-object v0
! 
!     const-wide v2, 0x2bc056ab0800006L
! 
!     invoke-static {v2, v3}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
! 
!     move-result-object v2
! 
!     invoke-interface {v0, v2}, Ljava/util/Map;->containsKey(Ljava/lang/Object;)Z
! 
!     move-result v0
! 
!     if-eqz v0, :cond_1
! 
!     goto :goto_0
! 
!     :cond_1
!     const/4 v1, 0x0
! 
!     :cond_2
!     :goto_0
!     return v1
  .end method
  
  .method public final isInitialized()Lrx/Observable;
--- 1318,1328 ----
  .end method
  
  .method public final isExperimentalAlpha$app_productionDiscordExternalRelease()Z
!     .locals 1
  
!     const/4 v0, 0x1
  
!     return v0
  .end method
  
  .method public final isInitialized()Lrx/Observable;
diff -crB com.discord-1078-base/smali/com/discord/widgets/chat/input/WidgetChatInputSend.smali com.discord-1078/smali/com/discord/widgets/chat/input/WidgetChatInputSend.smali
*** com.discord-1078-base/smali/com/discord/widgets/chat/input/WidgetChatInputSend.smali	2020-03-26 10:35:55.708906224 +0300
--- com.discord-1078/smali/com/discord/widgets/chat/input/WidgetChatInputSend.smali	2020-03-27 13:03:37.667100509 +0300
***************
*** 321,327 ****
      if-eqz p3, :cond_4
  
      .line 8
!     invoke-virtual {p3, p1}, Lcom/discord/app/AppActivity;->showKeyboard(Landroid/view/View;)V
  
      goto :goto_1
  
--- 321,329 ----
      if-eqz p3, :cond_4
  
      .line 8
!     # Without this line commented out, tapping on someone's avatar to mention them causes the keyboard to toggle
!     # Why? Because showKeyboard and hideKeyboard have the same function.
!     # invoke-virtual {p3, p1}, Lcom/discord/app/AppActivity;->showKeyboard(Landroid/view/View;)V
  
      goto :goto_1
  
diff -crB com.discord-1078-base/smali/com/discord/widgets/settings/WidgetSettingsAppearance$Model$Companion$get$1.smali com.discord-1078/smali/com/discord/widgets/settings/WidgetSettingsAppearance$Model$Companion$get$1.smali
*** com.discord-1078-base/smali/com/discord/widgets/settings/WidgetSettingsAppearance$Model$Companion$get$1.smali	2020-03-26 10:35:56.282242483 +0300
--- com.discord-1078/smali/com/discord/widgets/settings/WidgetSettingsAppearance$Model$Companion$get$1.smali	2020-03-27 12:53:46.243461585 +0300
***************
*** 87,97 ****
  
      if-nez p4, :cond_1
  
!     invoke-static {p1, v1}, Ly/u/b/j;->checkExpressionValueIsNotNull(Ljava/lang/Object;Ljava/lang/String;)V
! 
!     invoke-virtual {p1}, Lcom/discord/models/domain/ModelUser;->isStaff()Z
! 
!     move-result p4
  
      if-eqz p4, :cond_0
  
--- 87,93 ----
  
      if-nez p4, :cond_1
  
!     const/4 p4, 0x1
  
      if-eqz p4, :cond_0
  
diff -crB com.discord-1078-base/smali/com/discord/widgets/tabs/TabsFeatureFlag$fetchExperiment$2.smali com.discord-1078/smali/com/discord/widgets/tabs/TabsFeatureFlag$fetchExperiment$2.smali
*** com.discord-1078-base/smali/com/discord/widgets/tabs/TabsFeatureFlag$fetchExperiment$2.smali	2020-03-26 10:35:56.478910154 +0300
--- com.discord-1078/smali/com/discord/widgets/tabs/TabsFeatureFlag$fetchExperiment$2.smali	2020-03-27 12:53:46.243461585 +0300
***************
*** 74,82 ****
      move-result-object v0
  
      .line 3
!     invoke-virtual {p1}, Lcom/discord/stores/StoreExperiments$Experiment;->getBucket()I
! 
!     move-result p1
  
      const-string v1, "CACHE_KEY_TABS_EXPERIMENT_BUCKET"
  
--- 74,80 ----
      move-result-object v0
  
      .line 3
!     const/4 p1, 0x2
  
      const-string v1, "CACHE_KEY_TABS_EXPERIMENT_BUCKET"
  
diff -crB com.discord-1078-base/smali/com/discord/widgets/tabs/TabsFeatureFlag.smali com.discord-1078/smali/com/discord/widgets/tabs/TabsFeatureFlag.smali
*** com.discord-1078-base/smali/com/discord/widgets/tabs/TabsFeatureFlag.smali	2020-03-26 10:35:56.482243505 +0300
--- com.discord-1078/smali/com/discord/widgets/tabs/TabsFeatureFlag.smali	2020-03-27 12:53:46.243461585 +0300
***************
*** 74,80 ****
      .line 2
      iget-object p1, p0, Lcom/discord/widgets/tabs/TabsFeatureFlag;->sharedPreferences:Landroid/content/SharedPreferences;
  
!     const/4 v0, -0x1
  
      const-string v1, "CACHE_KEY_TABS_EXPERIMENT_BUCKET"
  
--- 74,80 ----
      .line 2
      iget-object p1, p0, Lcom/discord/widgets/tabs/TabsFeatureFlag;->sharedPreferences:Landroid/content/SharedPreferences;
  
!     const/4 v0, 0x2
  
      const-string v1, "CACHE_KEY_TABS_EXPERIMENT_BUCKET"
  
***************
*** 261,301 ****
  .end method
  
  .method public final isEnabled()Z
!     .locals 2
! 
!     .line 1
!     iget v0, p0, Lcom/discord/widgets/tabs/TabsFeatureFlag;->experimentBucket:I
! 
!     const/4 v1, 0x1
! 
!     if-lt v0, v1, :cond_0
! 
!     goto :goto_0
  
!     :cond_0
!     const/4 v1, 0x0
  
!     :goto_0
!     return v1
  .end method
  
  .method public final useOverlappingPanels()Z
!     .locals 2
! 
!     .line 1
!     iget v0, p0, Lcom/discord/widgets/tabs/TabsFeatureFlag;->experimentBucket:I
! 
!     const/4 v1, 0x2
! 
!     if-ne v0, v1, :cond_0
! 
      const/4 v0, 0x1
  
-     goto :goto_0
- 
-     :cond_0
-     const/4 v0, 0x0
- 
-     :goto_0
      return v0
  .end method
--- 261,276 ----
  .end method
  
  .method public final isEnabled()Z
!     .locals 1
  
!     const/4 v0, 0x1
  
!     return v0
  .end method
  
  .method public final useOverlappingPanels()Z
!     .locals 1
      const/4 v0, 0x1
  
      return v0
  .end method

